### DISCUSSION TITLE
Sorting by json values

### DISCUSSION THREAD

**Discussion by nsnayp13** [mikro-orm/mikro-orm#4830] [General]
When i try sorting by json number value it sorts like a string value

**To Reproduce**
Try to sort by numeric key in json field like this
`find(null,{ orderBy: { params: { age: 'desc' } }})`
query 
`select * from "item" order by "params"->>'age' desc`

**Expected behavior**
`select * from "item" order by "params"->'age' desc`

**Additional context**
->> means string
getSearchJsonPropertyKey function in PostgreSqlPlatform.js
By the way. filtering like 
`find({
params: {age :{$contains:[30]}}
})`
returns
`select * from item where "params"->>'age' @> '[30]'`
postgres syntax error

**Versions**

| Dependency | Version |
| - | - |
| node | 18 |
| typescript | 5.1.3 |
| mikro-orm | 5.8.1 |
| your-driver | postgres |

**Comment by B4nan** [comment]
> ->> means string

So the problem is in ordering numbers, namely putting things like `10` or `20` after `3` or `4`? Do you use `json` or `jsonb` columns?

I need to see more, you haven't provided almost anything. Try to put together a complete repro ideally.

> find({ params: {age :{$contains:[30]}} })

`$contains` is an array operator, it requires an array on both sides, you can't use it on a scalar value.
